### ISSUE TITLE
[bug] An explicit invalid --max-action-delta silently weakens guardrails instead of erroring

### ISSUE THREAD

**Issue by Tushar2604** [robocurve/inspect-robots#154]
**Describe the bug**

An invalid explicit `--max-action-delta` value silently weakens guardrails instead of failing. Instead of rejecting invalid user input, the CLI prints a warning to `stderr`, skips the delta limiter, continues with clamp-only guardrails, and exits successfully (`0`).

**Root cause:**
- `_build_guardrails` (`cli.py`) wraps `DeltaLimitApprover(space, max_delta=max_action_delta)` in a blanket `except ValueError`, and degrades to a warning for any failure.
- That degrade-per-component design is correct when a *derived* default can't apply to a space (for example, an embodiment declares no bounds), because there was never a working default to lose.
- However, when the user explicitly passes `--max-action-delta`, an invalid value (negative, zero, or non-finite) is a CLI input error, not an embodiment limitation. `DeltaLimitApprover` correctly rejects the value (`max_delta must be finite and > 0`), but that exception is caught by the same blanket handler and downgraded to a warning.
- As a result, the operator asked for a stricter per-step limit, mistyped it, and the run proceeds on real hardware with **less protection than explicitly requested**, unless they happen to notice the warning on `stderr`.

**To reproduce**

```console
inspect-robots run --task cubepick-reach --policy scripted --embodiment cubepick --max-action-delta -1
```

Or without the CLI:

```python
from inspect_robots.cli import main

main([
    "run",
    "--task", "cubepick-reach",
    "--policy", "scripted",
    "--embodiment", "cubepick",
    "--max-action-delta", "-1",
])
```

Output:

```text
guardrails warning: delta limit skipped: DeltaLimitApprover: max_delta must be finite and > 0
policy: scripted (--policy)
embodiment: cubepick (--embodiment)
guardrails: clamp
run status: completed
...
```

Exit code: `0`.

The run completes with **clamp-only** guardrails instead of the intended **clamp + delta-limit** chain configured by the flag.

**Expected behavior**

An explicitly invalid `--max-action-delta` (non-finite or `<= 0`) should fail immediately before any component resolution or hardware interaction, with a clear error such as:

```text
--max-action-delta must be finite and > 0, got -1.0
```

This should **not** change the existing degrade-per-component behavior for derived defaults. An embodiment with no declared bounds and no explicit `--max-action-delta` should continue warning and proceeding as it does today. Only an explicitly supplied invalid value should hard-fail.

**Environment**
- Inspect Robots version: `main` @ `7992f14` (`v0.21.1`, 4 commits past tag)
- Python version: 3.12.4
- OS: Windows 11 (also reproducible on Linux; this is argument validation, not platform-specific)
- NumPy version: 2.4.6
- Optional extras installed: none

**Additional context**

- `argparse` uses `type=float` for `--max-action-delta`, so the CLI value is always a `float | None`. From the CLI, the only reachable validation failures are non-finite values or values `<= 0`. The broadcast-shape validation in `DeltaLimitApprover._validate_max_delta()` is only reachable through the Python API by passing an array-like value directly.
- A narrow fix would be to validate `args.max_action_delta` in `_check_shared_run_conflicts()` (`cli.py`). That function already runs before component resolution for both `run` and `eval-set`, and already contains the sibling `--disable-guardrails` / `--max-action-delta` conflict check.
- No changes to `_build_guardrails` or `DeltaLimitApprover` itself should be necessary.
